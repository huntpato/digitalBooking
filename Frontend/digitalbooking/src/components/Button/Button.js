import React from "react";
import styles from "./Button.module.css"
import { Link } from "react-router-dom"


export const Button = (props) => {
  const { button, hide } = styles;
  const {text, link, handleClick} = props;
  let { visibility }= props;

  return (
    <Link to= {link || '/'} style={{textDecoration: 'none'}} onClick={ handleClick|| ''}> <button className={ visibility? hide: button}>
    {text}</button></Link> 
  );
};

