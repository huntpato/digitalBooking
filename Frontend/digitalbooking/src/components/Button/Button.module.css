.button {
  align-items: center;
  background-color: var(--background-white);
  border: 1.4px solid var(--principal-color);
  border-radius: 5px;
  color: var(--principal-color);
  display: flex;
  font-weight: 700;
  justify-content: center;
  margin-left: 15px;
  padding: 6px 30px 5px 30px;
  text-align: center;
  transition: 0.5s;
}

.button:active {
  box-shadow: 0px 4px 4px rgba(0, 0, 0, 0.15);
  transform: scale(0.97);
}

.button:hover {
  background-color: var(--principal-color);
  color: var(--background-white);
}

.hide{
  display: none !important;
  visibility:hidden;
}

@media (max-width: 768px) {
  .button {
    margin-left: 15px;
    padding: 6px 20px 5px 20px;
    transition: 0.5s;
  }
}
