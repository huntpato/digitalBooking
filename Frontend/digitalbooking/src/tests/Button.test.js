import React from 'react';
import { shallow } from 'enzyme';
import { Button } from '../components/Button/Button'
import '@testing-library/jest-dom'
import { expect } from '@jest/globals';

describe('Pruebas en componente <Button />', ()=> {

    const text = "Ver mas";
    const link = "/register";
    const visibility = false;
    const wrapper= shallow(<Button text={text} link= {link} visibility={visibility}/>);

    test('Comprobar que Button renderiza correctamente.', () => {
    
        expect( wrapper ).toMatchSnapshot();
      
    })

    test('Verificar que <Button /> muestra el texto enviado por props', () => {
        const button = wrapper.find('Link').text();
        expect( button ).toContain( text );
    }) 

    test('Verficar link del componente <Button />', () => {
        
        const button = wrapper.find('Link');
        expect( button.prop('to') ).toBe(link)
    })

    test('Verificar que el componente se puede visualizar cuando recibe como prop visibility= false', () => {
        
        const button = wrapper.find('button');
        expect(button.hasClass('button')).toBe(true);
    })
    
    test('Verificar que el componente no se esconda cuando reciba como prop visibility = false', () => {
        
        const button = wrapper.find('button');
        expect(button.hasClass('hide')).toBeFalsy();
    })
    


    
})
