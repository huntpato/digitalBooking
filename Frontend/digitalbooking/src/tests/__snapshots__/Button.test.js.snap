// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en componente <Button /> Comprobar que Button renderiza correctamente. 1`] = `
<Link
  onClick=""
  style={
    Object {
      "textDecoration": "none",
    }
  }
  to="/register"
>
   
  <button
    className="button"
  >
    Ver mas
  </button>
</Link>
`;
